cadence/test/flix_registry_test.cdc

```diff
@@ -0,0 +1,194 @@
+
+    let resolveScriptResult = executeScript(
+        "../scripts/resolve.cdc",
+        [REGISTRY_OWNER.address, cadenceBodyHash]
+    )
+
+    Test.expect(lookupScriptResult, Test.beSucceeded())
+
+    let resolvedFlix = lookupScriptResult.returnValue! as! FLIXRegistry.FLIX
+    Test.assertEqual(TEMPLATE_ID, resolvedFlix.id)
+    Test.assertEqual(jsonBody, resolvedFlix.jsonBody)
+    Test.assertEqual(cadenceBodyHash, resolvedFlix.cadenceBodyHash)
+    Test.assertEqual(FLIXRegistry.FLIXStatus.active, resolvedFlix.status)
+}
+
+access(all)
+fun testShouldLinkAlias() {
+    let txResult = executeTransaction(
+        "../transactions/link_alias.cdc",
+        [NEW_ALIAS, TEMPLATE_ID],
+        REGISTRY_OWNER
+    )
+    Test.expect(txResult, Test.beSucceeded())
+
+    let typ = Type<FLIXRegistry.AliasLinked>()
+    Test.assertEqual(1, Test.eventsOfType(typ).length)
+
+    let scriptResult = executeScript(
+        "../scripts/get_all_alias.cdc",
+        [REGISTRY_OWNER.address]
+    )
+    Test.expect(scriptResult, Test.beSucceeded())
+
+    let aliases = scriptResult.returnValue! as! {String: String}
+
+    Test.assertEqual(TEMPLATE_ID, aliases[ALIAS]!)
+    Test.assertEqual(TEMPLATE_ID, aliases[NEW_ALIAS]!)
+}
+
+access(all)
+fun testShouldUnlinkAlias() {
+    let txResult = executeTransaction(
+        "../transactions/unlink_alias.cdc",
+        [ALIAS],
+        REGISTRY_OWNER
+    )
+    Test.expect(txResult, Test.beSucceeded())
+
+    let typ = Type<FLIXRegistry.AliasUnlinked>()
+    Test.assertEqual(1, Test.eventsOfType(typ).length)
+
+    let scriptResult = executeScript(
+        "../scripts/get_all_alias.cdc",
+        [REGISTRY_OWNER.address]
+    )
+    Test.expect(scriptResult, Test.beSucceeded())
+
+    let aliases = scriptResult.returnValue! as! {String: String}
+
+    Test.assertEqual(nil, aliases[ALIAS])
+    Test.assertEqual(TEMPLATE_ID, aliases[NEW_ALIAS]!)
+}
+
+access(all)
+fun testShouldDeprecateFlixWithTemplateId() {
+    let lookupScriptResultBefore = executeScript(
+        "../scripts/lookup.cdc",
+        [REGISTRY_OWNER.address, TEMPLATE_ID]
+    )
+    let flixBefore = lookupScriptResultBefore.returnValue! as! FLIXRegistry.FLIX
+    Test.assertEqual(FLIXRegistry.FLIXStatus.active, flixBefore.status)
+
+    let txResult = executeTransaction(
+        "../transactions/deprecate_flix.cdc",
+        [TEMPLATE_ID],
+        REGISTRY_OWNER
+    )
+    Test.expect(txResult, Test.beSucceeded())
+
+    let typ = Type<FLIXRegistry.Deprecated>()
+    Test.assertEqual(1, Test.eventsOfType(typ).length)
+
+    let lookupScriptResultAfter = executeScript(
+        "../scripts/lookup.cdc",
+        [REGISTRY_OWNER.address, TEMPLATE_ID]
+    )
+
+    Test.expect(lookupScriptResultAfter, Test.beSucceeded())
+
+    let flixAfter = lookupScriptResultAfter.returnValue! as! FLIXRegistry.FLIX
+    Test.assertEqual(FLIXRegistry.FLIXStatus.deprecated, flixAfter.status)
+}
+
+access(all)
+fun testShouldRemoveFlix() {
```

Comment:
Todo: Ask Peter if tests can be run independent of each other

Reply:
There is no built-in way to run test cases in isolation, as there is only one blockchain under the hood. However, this can be achieved by creating/loading snapshots of blockchain state. As is the case here. Furthermore, to make sure that flaky tests don't go unnoticed, there is a flag (`--random`) for random test case execution, e.g.:
```bash
flow test --cover --covercode="contracts" --random cadence/test/flix_registry_test.cdc
```